context.c

```diff
@@ -237,7 +237,8 @@ struct iio_device * iio_context_find_device(const struct iio_context *ctx,
 	for (i = 0; i < ctx->nb_devices; i++) {
 		struct iio_device *dev = ctx->devices[i];
 		if (!strcmp(dev->id, name) ||
-				(dev->name && !strcmp(dev->name, name)))
+				(dev->name && !strcmp(dev->name, name)) ||
+				(dev->label && !strcmp(dev->label, name)))
```

Comment:
I'm not sure about the device id. But between the label and the name, I would first match against the label. If a label is present, most likely it is how the device wants to be found 😄

Reply:
I would disagree here. The name is still the main way to identify a device. A label comes into play when you have multiple devices with the same name and you would like to know which is which. That is just the way I used labels.

Reply:
Yes, exactly. And for me that means that if someone defines a label in DTS for a device, he wants it to be primarily identified by that label. Image the following scenario where someone has two devices with the part name foo. It would be a valid option to just use a label in the second device like foo_1, leaving the first device just with the name. With this, I guess `iio_context_find_device()` could always return the same device instead of foo and foo_1...

Reply:
@mirceacaprioru if you have multiple devices with the same name, then the label is the differenciating factor. Therefore it has a higher priority than the name, and @nunojsa is right.

Reply:
ack. Thanks